oap-ape/ape-native/src/test/CMakeLists.txt

```diff
@@ -43,31 +43,114 @@ add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/googletest-src
 
 set(CMAKE_CXX_STANDARD 17)
 
+LINK_DIRECTORIES(/lib /lib64 /usr/local/lib /usr )
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+find_package(Arrow REQUIRED)
+
 add_executable(convertorTest convertorTest.cc)
-target_link_libraries(convertorTest gtest_main parse)
+target_link_libraries(convertorTest gtest_main parse
+    -Wl,-Bstatic
+    -Wl,--whole-archive
+    arrow_static
+    -Wl,--no-whole-archive
+    -Wl,-Bdynamic
+    -Wl,--as-needed
+    Threads::Threads
+    -Wl,--no-as-needed)
 add_test(NAME convertorTest COMMAND ${EXECUTABLE_OUTPUT_PATH}/convertorTest)
 
 add_executable(decimalTest decimalTest.cc)
-target_link_libraries(decimalTest gtest_main parquet_jni)
+target_link_libraries(decimalTest gtest_main parquet_jni
+    -Wl,-Bstatic
+    -Wl,--whole-archive
+    -Wl,--start-group
+    arrow_static
+    libhiredis.a
+    libredis++.a
+    libplasma.a
+    libparquet.a
+    -Wl,--end-group
+    -Wl,--no-whole-archive
+    -Wl,-Bdynamic
+    -Wl,--as-needed
+    Threads::Threads
+    -Wl,--no-as-needed)
 add_test(NAME decimalTest COMMAND ${EXECUTABLE_OUTPUT_PATH}/decimalTest)
 
 add_executable(aggParserTest aggParserTest.cc)
-target_link_libraries(aggParserTest gtest_main parquet_jni)
+target_link_libraries(aggParserTest gtest_main parquet_jni
+   -Wl,-Bstatic
```

Comment:
could these parameters become one(or more) cmake variable to reduce repeat code?

Reply:
yes, changed them.